From 93b71573f0a9966c572857d0ff5905d17b368d8c Mon Sep 17 00:00:00 2001
From: akodanka <anoob.anto.kodankandath@intel.com>
Date: Fri, 17 Jan 2020 00:38:30 +0530
Subject: [PATCH] WA Patch for realtek: Set runtime autosuspend delay

Mic detection event is missed on plug in of headset with mic.

As auto runtime suspend delay is not set for realtek codec from SOF,
the codec suspend is called immediately after reporting of headphone event.
Due to this, the following mic event is missed out.

Set autosuspend delay to 3s to address mic detection issue seen
with realtek 285 codec.

Signed-off-by: Anoob Anto Kodankandath <anoob.anto.kodankandath@intel.com>
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Praveen Diwakar <praveen.diwakar@intel.com>
---
 sound/pci/hda/patch_realtek.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/sound/pci/hda/patch_realtek.c b/sound/pci/hda/patch_realtek.c
index 5673efd6d5df..ef7d289f3226 100644
--- a/sound/pci/hda/patch_realtek.c
+++ b/sound/pci/hda/patch_realtek.c
@@ -41,6 +41,11 @@
 /* keep halting ALC5505 DSP, for power saving */
 #define HALT_REALTEK_ALC5505
 
+#ifdef CONFIG_PM
+/* time in ms for runtime suspend delay */
+#define REALTEK_SUSPEND_DELAY_MS	3000
+#endif // CONFIG_PM
+
 /* extra amp-initialization sequence types */
 enum {
 	ALC_INIT_UNDEFINED,
@@ -7896,6 +7901,9 @@ static int patch_alc269(struct hda_codec *codec)
 		spec->shutup = alc225_shutup;
 		spec->init_hook = alc225_init;
 		spec->gen.mixer_nid = 0;
+#ifdef CONFIG_PM
+		snd_hda_set_power_save(codec->bus, REALTEK_SUSPEND_DELAY_MS);
+#endif // CONFIG_PM
 		break;
 	case 0x10ec0225:
 	case 0x10ec0295:
-- 
2.24.0

